include/zephyr/drivers/rtc.h

```diff
@@ -0,0 +1,527 @@
+/*
+ * Copyright (c) 2023 Trackunit Corporation
+ * Copyright (c) 2023 Bjarki Arge Andreasen
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file drivers/rtc.h
+ * @brief Public real time clock driver API
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_RTC_H_
+#define ZEPHYR_INCLUDE_DRIVERS_RTC_H_
+
+/**
+ * @brief RTC Interface
+ * @defgroup rtc_interface RTC Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#include <zephyr/types.h>
+#include <zephyr/device.h>
+#include <errno.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Mask for alarm time fields to enable when setting alarm time
+ * @{
+ */
+#define RTC_ALARM_TIME_MASK_SECOND      BIT(0)
```

Comment:
I have yet to find an RTC alarm that supports the subsecond field, many don't even support the second field, so it is left out for now at least

Reply:
I have not either.  They have all implicitly matched subseconds == 0 in their alarms.  I don't think I've ever seen one that can do YEARDAY either for that matter.

But suppose one exists?  It wouldn't be possible to add it later.  Because every alarm should set rtc.tm_nsec to zero and specify if it wants to occur at zero fractional seconds (which is all every chip supports).  When this new chip that can trigger at tm_nsec = 5000000 wants to use it and adds RTC_ALARM_TIME_MASK_NSEC, all existing code will break because none of it is using this new flag.

Adding it makes the mask consistent with the rtc_time struct, one mask bit per field.  But one always has to set the mask, that is annoying.  So how about:

```
There is no RTC_ALARM_TIME_MASK_NSEC bit because it is always implicitly enabled, i.e. the alarm will only trigger when the time matches the tm_nsec field specified.  Most, if not all, RTC devices only support triggering when tm_nsec == 0, so the tm_nsec field should be set to zero when setting an alarm.  Which is to say, they trigger at the start of the second in which the alarm is to occur.  
```
This heads off app authors leaving the tm_nsec field at a random value when setting an alarm because it doesn't matter on their hardware, then throwing a field when their code breaks when someone tries to use the field.